[BBCSPA-1446] Checkbox position change and more

Tuxosaurus opened the pull request:
Add a prop to invert text and checkbox, defaults to checkbox before text.
Prevent empty nodes when labels are missing.
Unify stories into one entry.

pith commented on CHANGELOG.md:
```diff
@@ -1,5 +1,6 @@
 # Unreleased
 - [timePicker] Fix display on small devices
+- [Checkbox] Reverse label and input and more improvements
```
`Reverse label and input and made it configurable`

Tuxosaurus replied:
Done

pith commented on src/checkbox/index.tsx:
```diff
@@ -11,6 +11,7 @@ export interface CheckboxProps {
   readonly value?: string,
   readonly checked?: boolean,
   readonly disabled?: boolean,
+  readonly labelBeforeInput?: boolean,
```
I would prefer `labelPosition` with an enum left / right

Tuxosaurus replied:
I'll do that

Tuxosaurus replied:
Done

pith commented on src/checkbox/index.tsx:
```diff
@@ -58,9 +60,14 @@ export default class Checkbox extends PureComponent <CheckboxProps, CheckboxStat
   render() {
-    const { className, name, children, value, subLabel } = this.props
+    const { className, name, children, value, subLabel, labelBeforeInput } = this.props
+    const hasLabel = children || subLabel
```
Is there a design where we can have a checkbox without label. Also is it really possible to have a sublabel without label ?

Tuxosaurus replied:
I don't know, and hope that we never have a checkbox without a label, it's only a little change to make sure we don't serve empty markup in this event.
Having a subLabel only is possible with the current implementation so I didn't challenge it.

pith commented on src/checkbox/index.unit.tsx:
```diff
@@ -77,3 +77,21 @@ it('should trigger a change event on an async checkbox', () => {
   checkbox.find('input').simulate('change')
 })
+
+it('should display the label on the left and the checkbox on the right', () => {
```
nit: you could do a snapshot test

Tuxosaurus replied:
Done

loick commented on src/checkbox/index.tsx:
```diff
@@ -58,9 +65,19 @@ export default class Checkbox extends PureComponent <CheckboxProps, CheckboxStat
   render() {
-    const { className, name, children, value, subLabel } = this.props
+    const { className, name, children, value, subLabel, labelPosition } = this.props
+    const hasLabel = children || subLabel
```
hasLabel is always true? 'cause children seems to be mandatory

loick replied:
children.length > 0 || (subLabel && subLabel.length > 0)

nickoola commented:
Can you add screenshots of both versions ?

Zarthia commented:
![screen shot 2018-06-18 at 10 45 01](https://user-images.githubusercontent.com/3199901/41526506-c1728992-72e4-11e8-95f4-fc0880266200.png)
Label should be placed right after the checkbox (with the adequate margin)

![screen shot 2018-06-18 at 10 48 18](https://user-images.githubusercontent.com/3199901/41526661-2caa3a02-72e5-11e8-8f05-16326327bd07.png)
Also, when you want to display the checkbox only, without label/sublabel, I guess we shouldn't have the `<label>` element

Tuxosaurus commented:
Changed "Position" to "Display" as per @Zarthia's request
Added a "None" display that preserves the existing label structure but makes it "visually hidden"

<img width="422" alt="checkbox_label_left" src="https://user-images.githubusercontent.com/373381/41536145-0c75b070-7305-11e8-958f-0e4f4e8f92cb.png">
<img width="424" alt="checkbox_label_right" src="https://user-images.githubusercontent.com/373381/41536147-0ca097ea-7305-11e8-94ea-e34f312e10d9.png">
<img width="423" alt="checkbox_label_none" src="https://user-images.githubusercontent.com/373381/41536146-0c8b764e-7305-11e8-8eb9-7cabccd7f0ff.png">

nickoola commented:
Just try your update localy on Kiaros before merge.

Tuxosaurus commented:
Checked on kairos (local), change is OK and doesn't need any adjustment
<img width="313" alt="screen shot 2018-06-19 at 13 47 57" src="https://user-images.githubusercontent.com/373381/41596712-7988c0a0-73cb-11e8-8fda-f46a9ffe852a.png">
<img width="311" alt="screen shot 2018-06-19 at 14 02 14" src="https://user-images.githubusercontent.com/373381/41596713-79af1ab6-73cb-11e8-9970-d852809e5a31.png">
